tests/e2e/e2e_utils.py

```diff
@@ -21,19 +24,33 @@ def run_oneshot_for_e2e_testing(
 ):
     # Load model.
     oneshot_kwargs = {}
-    loaded_model = AutoModelForCausalLM.from_pretrained(
-        model, device_map=device, torch_dtype="auto"
+    pretrained_model_class = get_model_class(model_class)
+    loaded_model = pretrained_model_class.from_pretrained(
+        model,
+        device_map=device,
+        torch_dtype="auto",
+        trust_remote_code=True,
+        _attn_implementation="eager",
     )
-    tokenizer = AutoTokenizer.from_pretrained(model)
+    processor = AutoProcessor.from_pretrained(model, trust_remote_code=True)
 
     if dataset_id:
         ds = load_dataset(dataset_id, name=dataset_config, split=dataset_split)
         ds = ds.shuffle(seed=42).select(range(num_calibration_samples))
-        ds = preprocess_tokenize_dataset(ds, tokenizer, max_seq_length)
+        ds = process_dataset(ds, processor, max_seq_length)
         oneshot_kwargs["dataset"] = ds
         oneshot_kwargs["max_seq_length"] = max_seq_length
         oneshot_kwargs["num_calibration_samples"] = num_calibration_samples
 
+        # TODO better conditional on when multimodal data-collator should be added
```

Comment:
taking suggestions for a better way to handle this, or I can just drop the TODO

Reply:
I think this is fine. All multimodal models can be used with text-only, but in those cases you don't need the custom collator.

Therefore, using the custom collator is only dependent on there being multimodal input data